packages/controls/src/controls/shape-v2/shape-v2.ts

```diff
@@ -0,0 +1,322 @@
+import { z } from 'zod'
+
+import { mapValues } from '../../lib/functional'
+import { safeParse, type ParseResult } from '../../lib/zod'
+
+import { ControlDataTypeKey, type Data } from '../../common'
+import {
+  type CopyContext,
+  type MergeTranslatableDataContext,
+} from '../../context'
+import { type IntrospectionTarget } from '../../introspection'
+import {
+  SerializationSchema,
+  type DeserializedRecord,
+  type SerializedRecord,
+} from '../../serialization'
+
+import {
+  type DataType as DataType_,
+  type ResolvedValueType as ResolvedValueType_,
+  type ValueType as ValueType_,
+} from '../associated-types'
+import { ControlDefinition, serialize, type SchemaType } from '../definition'
+import { type SendMessage } from '../instance'
+
+import { ShapeV2Control } from './shape-v2-control'
+
+type ItemDefinition = ControlDefinition<string, unknown, any, any, any>
+type KeyDefinitions = Record<string, ItemDefinition>
+
+export type Config<Defs extends KeyDefinitions = KeyDefinitions> = {
+  label?: string
+  layout?: typeof Definition.Layout.Inline | typeof Definition.Layout.Popover
+  readonly type: Defs
+}
+
+export type ShapeV2DataTypeV0<C extends Config> = {
+  [K in keyof C['type']]?: DataType_<C['type'][K]>
+}
+
+export const ShapeV2DataTypeV1VersionId = 'shape-v2::v1' as const
```

Comment:
I would not pollute the global scope with this; I think we can just say `typeof Definition.v1DataType` below.

Reply:
Ok. I need this value in cosmos, and `v1DataType` is private in the schema. Are you ok with me exposing it?

Reply:
We can make `v1DataType` public.  I think ideally we want to find a way to avoid checking it directly outside of the control implementation, though.

Reply:
In Cosmos, we are using the `version` to differentiate data. Should I be doing something similar?

![CleanShot 2024-12-20 at 13 26 28@2x](https://github.com/user-attachments/assets/25f842a6-1fd9-4164-889f-25a3e77ca0fa)

Reply:
Actually that is for the version of the image control

Reply:
I don't think we have any references in leo to actual data versions.

Reply:
Good news. With a `versionData` static function this is no longer required.